src/com/mendix/widget/ProgressCircle/components/__tests__/ProgressCircle.spec.ts

```diff
@@ -0,0 +1,266 @@
+                DOM.div({className: "widget-progress-circle-medium", onClick: jasmine.any(Function) as any })
+            )
+        );
+    });
+
+    it("creates a progress circle", () => {
+        spyOnCircle();
+        const progress = newCircleInstance({ value: 80 });
+
+        progress.componentDidMount();
+
+        expect(progressbar.Circle).toHaveBeenCalled();
+    });
+
+    it("sets the progress percentage", () => {
+        spyOn(progressbar.Circle.prototype, "setText").and.callThrough();
+        const setText = progressbar.Circle.prototype.setText as jasmine.Spy;
+        spyOnCircle();
+
+        const progress = newCircleInstance({ animate: false, value: 80 });
+        progress.componentDidMount();
+
+        expect(setText).toHaveBeenCalled();
+    });
+
+    it("updates the progress percentage when the values are changed", () => {
+        spyOn(progressbar.Circle.prototype, "setText").and.callThrough();
+        const setText = progressbar.Circle.prototype.setText as jasmine.Spy;
+        spyOnCircle();
+
+        const progress = newCircleInstance({ value: 80 });
+        progress.componentDidMount();
+        progress.componentDidUpdate();
+
+        expect(setText).toHaveBeenCalledTimes(2);
+    });
+
+    it("destroys progress circle on unmount", () => {
+        spyOn(progressbar.Circle.prototype, "destroy").and.callThrough();
+        const destroy = progressbar.Circle.prototype.destroy as jasmine.Spy;
+        spyOnCircle();
+
+        const progress = newCircleInstance({ value: 80 });
+        progress.componentDidMount();
+        progress.componentWillUnmount();
+
+        expect(destroy).toHaveBeenCalled();
+    });
+
+    it("renders a circle no text when no value is specified", () => {
+        spyOnCircle();
+
+        const progress = newCircleInstance({ value: null });
+        progress.componentDidMount();
+
+        expect(progressCircle.text.textContent).toBe("--");
+    });
+
+    it("renders a circle with the text set to Invalid when the maximum value is less than 1", () => {
+        spyOnCircle();
+
+        const progress = newCircleInstance({
+            animate: false,
+            maximumValue: -1,
+            onClickType: "doNothing",
+            value: 80
+        });
+        progress.componentDidMount();
+
+        expect(progressCircle.text.textContent).toBe("Invalid");
+    });
+
+    it("renders a circle with negative values when the value is less than 0", () => {
+        spyOnCircle();
+
+        const progress = newCircleInstance({ animate: false, value: -1 });
+        progress.componentDidMount();
+
+        expect(progressCircle.text.textContent).toBe("-1%");
+    });
+
+    it("renders a circle with text greater than 100% when the value is greater than the maximum", () => {
+        spyOnCircle();
+
+        const progress = newCircleInstance({ value: 180 });
+        progress.componentDidMount();
+
+        expect(progressCircle.text.textContent).toBe("180%");
```

Comment:
Couldn't test this. Regardless of the value, `progressCircle.value()` returned `NaN`.